LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -576,14 +579,17 @@ void ProcessAndSendMacCommands(LoRaPayloadData payloadData, ref Dictionary<strin
             uint payloadFcnt,
             ILoRaDeviceFrameCounterUpdateStrategy frameCounterStrategy)
         {
-            if (fcntDown > 0)
+            if (fcntDown.HasValue)
+            {
+                // if it was previously calculated, use that value
                 return fcntDown.Value;
+            }
 
             var newFcntDown = await frameCounterStrategy.NextFcntDown(loRaDevice, payloadFcnt);
 
             // Failed to update the fcnt down
             // In multi gateway scenarios it means the another gateway was faster than using, can stop now
-            if (newFcntDown <= 0)
+            if (newFcntDown > 0)
```

Comment:
Shouldn't this be the opposite? If the newFcntDown > 0 means we have a valid frame down counter. If it is <= 0 then is invalid.

Reply:
Good catch. Thanks. Would have resulted in a wrong debug message.